python_modules/dagster/dagster/_core/secrets/env_file.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import os
+from typing import Dict, Optional
+
+from dotenv import dotenv_values
+
+import dagster._check as check
+from dagster._config import Field, StringSource
+from dagster._serdes import ConfigurableClass
+
+from .loader import SecretsLoader
+
+
+class EnvFileLoader(SecretsLoader, ConfigurableClass):
```

Comment:
I'm curious how this will interact with other popular env loaders. For example, when you cd into a directory with a .env file while using ZSH's dotenv plugin, you get prompted with:

```
dotenv: found '.env' file. Source it? ([Y]es/[n]o/[a]lways/n[e]ver)
```

Will we respect those decisions? If I told my shell not to source my .env file, I'd be surprised if dagster automatically sourced it anyway.

Should we prompt before loading? Should we include CLI flags to disable loading?

Reply:
"If I told my shell not to source my .env file, I'd be surprised if dagster automatically sourced it anyway." - i'm not sure i agree with this, if you have a .env file in your dagster folder, it seems plausible to me that dagster would use it.

I'm not totally opposed to adding a flag to disable it but it seems like the kind of thing i'd wait for somebody to ask for - not hard to change it later if it comes up.